examples/network-container/container-peering-cluster.yml

```diff
@@ -0,0 +1,117 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: >-
+  This template creates a cluster on the MongoDB Atlas API and set the Network Peering.
+Metadata: 
+  AWS::CloudFormation::Interface: 
+    ParameterGroups: 
+      - Label: 
+          default: "Cluster Configurations"
+        Parameters:
+          - ClusterName
+      - Label: 
+          default: "VPC Peering Configuration"
+        Parameters:
+          - VpcId
+          - AwsAccountId
+          - AwsRegionName
+          - RouteTableCidrBlock
+Parameters:
+  ClusterName:
+    Type: String
+    Description: Name of the Atlas Cluster
+  ClusterRegionName:
+    Type: String
+    Description: Region of the Atlas Cluster
+  AtlasCidrBlock:
+    Type: String
+    Description: CIDR block of the Atlas VPC
+  VpcId:
+    Type: String
+    Description: Unique string that identifies the VPC on Amazon Web Services (AWS) that you want to peer with the MongoDB Cloud VPC.
+  AwsAccountId:
+    Type: String
+    Description: Unique twelve-digit string that identifies the Amazon Web Services (AWS) account that owns the VPC that you peered with the MongoDB Cloud VPC.
+  AwsRegionName:
+    Type: String
+    Description: Amazon Web Services (AWS) region where the Virtual Peering Connection (VPC) that you peered with the MongoDB Cloud VPC resides. The resource returns null if your VPC and the MongoDB Cloud VPC reside in the same region.
+  RouteTableCidrBlock:
+    Type: String
+    Description: Internet Protocol (IP) addresses expressed in Classless Inter-Domain Routing (CIDR) notation of the VPC's subnet that you want to peer with the MongoDB Cloud VPC.
+  Profile:
+    Type: String
+    Description: "Secret Manager Profile that contains the Atlas Programmatic keys."
+  ProjectId:
+    Type: String
+    Description: "The ID of your atlas project."
+
+Mappings: {}
+Resources:
+  Cluster:
+    Type: 'MongoDB::Atlas::Cluster'
+    DependsOn: NetworkContainer # Important to wait for the Network Container, otherwise the network container deletion will fail when deleting the stack
```

Comment:
added example with cluster

Reply:
[q] a little confused here, my understanding is basically Cluster should be deleted first and once that's done we can proceed with container deletion, is that right?
If yes, shouldn't it be that NetworkContainer resource 'dependsOn' Cluster resource?
Although I do see the important note here https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-dependson.html , wanted to confirm

Reply:
Using the link you provided as an example:
resource A is the cluster and resource B is the network container -> Resource A is deleted before resource B.

```
For example, if the properties of resource A use a !Ref to resource B, the following rules apply:

Resource B is created before resource A.
Resource A is deleted before resource B.
Resource B is updated before resource A.
```